roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -153,7 +168,15 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
                                     ),
                                 ),
                             )))? as usize;
-                    transactions.insert(index, tx);
+                    if index >= transactions.len() {
```

Comment:
From doc.rs
```
let mut vec = vec!["hello"];
vec.resize(3, "world");
assert_eq!(vec, ["hello", "world", "world"]);
```
This is the vector for the transactions that will be mined, in this order. I think that this is used to reconstruct the block. If some transaction is duplicated, this may lead propagation of invalid block. Be careful!
I suggest not resizing, as, in fact, it is already performed by `insert()` method few lines below.